src/ruleset/formats.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable security/detect-unsafe-regex */
+
+import { isObject } from '../utils';
+
+import type { Format } from '@stoplight/spectral-core';
+import type { MaybeAsyncAPI } from '../types';
+
+const aas2Regex = /^2\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$/;
+const aas2_0Regex = /^2\.0(?:\.[0-9]*)?$/;
+const aas2_1Regex = /^2\.1(?:\.[0-9]*)?$/;
+const aas2_2Regex = /^2\.2(?:\.[0-9]*)?$/;
+const aas2_3Regex = /^2\.3(?:\.[0-9]*)?$/;
+const aas2_4Regex = /^2\.4(?:\.[0-9]*)?$/;
+const aas2_5Regex = /^2\.5(?:\.[0-9]*)?$/;
+const aas2_6Regex = /^2\.6(?:\.[0-9]*)?$/;
+
+const isAas2 = (document: unknown): document is { asyncapi: string } & Record<string, unknown> =>
+  isObject(document) && 'asyncapi' in document && aas2Regex.test(String((document as MaybeAsyncAPI).asyncapi));
+
+export const aas2: Format = isAas2;
+aas2.displayName = 'AsyncAPI 2.x';
+
+export const aas2_0: Format = (document: unknown): boolean =>
+  isAas2(document) && aas2_0Regex.test(String((document as MaybeAsyncAPI).asyncapi));
+aas2_0.displayName = 'AsyncAPI 2.0.x';
+
+export const aas2_1: Format = (document: unknown): boolean =>
+  isAas2(document) && aas2_1Regex.test(String((document as MaybeAsyncAPI).asyncapi));
+aas2_1.displayName = 'AsyncAPI 2.1.x';
+
+export const aas2_2: Format = (document: unknown): boolean =>
+  isAas2(document) && aas2_2Regex.test(String((document as MaybeAsyncAPI).asyncapi));
+aas2_2.displayName = 'AsyncAPI 2.2.x';
+
+export const aas2_3: Format = (document: unknown): boolean =>
+  isAas2(document) && aas2_3Regex.test(String((document as MaybeAsyncAPI).asyncapi));
+aas2_3.displayName = 'AsyncAPI 2.3.x';
+
+export const aas2_4: Format = (document: unknown): boolean =>
+  isAas2(document) && aas2_4Regex.test(String((document as MaybeAsyncAPI).asyncapi));
+aas2_4.displayName = 'AsyncAPI 2.4.x';
+
+export const aas2_5: Format = (document: unknown): boolean =>
+  isAas2(document) && aas2_5Regex.test(String((document as MaybeAsyncAPI).asyncapi));
+aas2_5.displayName = 'AsyncAPI 2.5.x';
+
+export const aas2_6: Format = (document: unknown): boolean =>
```

Comment:
I know, but it will be release in two (max) days 😄

Reply:
Right, but just keep in mind that while it is likely that the version is introduced, you wouldn't be able to revert it later, because it's part of a larger commit.

So new features = new PR, at least in theory 😆

Reply:
So I will wait for it :P Don't worry.